core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2413,7 +2413,7 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
               Pull.output(acc) >> go(hd, size, tl)
```

Comment:
I think this is another error. The `size` should ideally be the desired size of the next output chunk, but that is the one being emited in `acc`. Instead should be -1, so that the next iteration recomputes a new size based on the first chunk of `tl`.

Reply:
Yes! I realize it now.